mmv1/third_party/terraform/services/storage/resource_storage_folder.go.tmpl

```diff
@@ -0,0 +1,343 @@
+		RetryFunc: func() error {
+			res, err = config.NewStorageClient(userAgent).Folders.Insert(bucket, folder).Do()
+			return err
+		},
+		Timeout:              d.Timeout(schema.TimeoutCreate),
+		ErrorRetryPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.Is429RetryableQuotaError},
+	})
+
+	log.Printf("[DEBUG] Creating new Folder: %#v", folder.Name)
+	if err != nil {
+		return fmt.Errorf("Error creating Folder: %s", err)
+	}
+
+	// Store the ID now
+	d.SetId(res.Id)
+
+	if err := d.Set("force_destroy", forceDestroy); err != nil {
+		return fmt.Errorf("Error setting force destroy flag: %s", err)
+	}
+
+	log.Printf("[DEBUG] Finished creating Folder %q: %#v", d.Id(), res)
+
+	return resourceStorageFolderRead(d, meta)
+}
+
+func resourceStorageFolderRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+	bucket := d.Get("bucket").(string)
+	folder := d.Get("name").(string)
+	var res *storage.Folder
+	err = transport_tpg.Retry(transport_tpg.RetryOptions{
+		RetryFunc: func() (operr error) {
+			var retryErr error
+			res, retryErr = config.NewStorageClient(userAgent).Folders.Get(bucket, folder).Do()
+			return retryErr
+		},
+		Timeout:              d.Timeout(schema.TimeoutRead),
+		ErrorRetryPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.IsNotFoundRetryableError("folder read")},
+	})
+	if err != nil {
+		return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("StorageFolder %q", d.Id()))
+	}
+
+	if err := d.Set("create_time", flattenStorageFolderCreateTime(res.CreateTime, d, config)); err != nil {
+		return fmt.Errorf("Error reading Folder: %s", err)
+	}
+	if err := d.Set("update_time", flattenStorageFolderUpdateTime(res.UpdateTime, d, config)); err != nil {
+		return fmt.Errorf("Error reading Folder: %s", err)
+	}
+	if err := d.Set("metageneration", flattenStorageFolderMetageneration(res.Metageneration, d, config)); err != nil {
+		return fmt.Errorf("Error reading Folder: %s", err)
+	}
+	if err := d.Set("bucket", flattenStorageFolderBucket(res.Bucket, d, config)); err != nil {
+		return fmt.Errorf("Error reading Folder: %s", err)
+	}
+	if err := d.Set("name", flattenStorageFolderName(res.Name, d, config)); err != nil {
+		return fmt.Errorf("Error reading Folder: %s", err)
+	}
+	if err := d.Set("self_link", tpgresource.ConvertSelfLinkToV1(res.SelfLink)); err != nil {
+		return fmt.Errorf("Error reading Folder: %s", err)
+	}
+
+	return nil
+}
+
+func resourceStorageFolderUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	_ = config
+
+	// we can only get here if force_destroy was updated
+	if d.Get("force_destroy") != nil {
+		if err := d.Set("force_destroy", d.Get("force_destroy")); err != nil {
+			return fmt.Errorf("Error updating force_destroy: %s", err)
+		}
```

Comment:
Not sure but if the field is not being pass down to the API request then do we need to explicitly set in updater?

Reply:
not passing to API but need to manage in tf state file, customers may switch from false to true and vice versa, based on this flag we manage the deletion logic when destroying the resource